Add HelperOptions section

jf-viguier opened the pull request:
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Description?  | Add HelperOptions section to explain how to manage configuration form in modules.
| Fixed ticket? | none

<!-- Click the form's "Preview" button to make sure the table is functional in GitHub. Thank you! -->

Commit:
Precise nodejs version and how to switch

Commit:
Add node-gypp and windows-build-tools precisions

jf-viguier committed:
Add HelperOptions section

Progi1984 commented on src/content/1.7/modules/creation/adding-configuration-page.md:
```diff
@@ -320,3 +320,100 @@ in the back office:
+Using HelperOptions
+----------------
+
+THe `HelperOptions` helper is the recommanded one for managing configurations i.e. values stored in the configuration table.
```
Where is it recommended ?

jf-viguier replied:
it's recommended by @Matt75

jf-viguier replied:
@Progi1984 you can see it there in action https://github.com/Matt75/displayordercarrier/blob/master/controllers/admin/AdminDisplayOrderCarrierController.php

kpodemski replied:
I wouldn't say that it is a recommended way as it's legacy part of the code, it is for legacy part, but for sure not for a modern architecture :) we need to inform developers about it

jf-viguier replied:
Thanks @kpodemski I've changed this.

Progi1984 commented on src/content/1.7/modules/creation/adding-configuration-page.md:
```diff
@@ -320,3 +320,100 @@ in the back office:
+    ['description'] => $this->l('Display as description'),         // Displays an informational box above the fields.
+    ['info'] => $this->l('Display as info'),                       // Displays an unstyled text above the fields.
+    'fields' => array(                                             // The various option fields.
+      'PS_CARRIER_DEFAULT' => array(                               // The aray is named after the option's ID. It must be the
```
```suggestion
      'PS_CARRIER_DEFAULT' => array(                               // The array is named after the option's ID. It must be the
```

Progi1984 commented on src/content/1.7/modules/creation/adding-configuration-page.md:
````diff
@@ -320,3 +320,100 @@ in the back office:
+],
+```
+
+### checkbox example
````
```suggestion
### Checkbox example
```

Progi1984 commented on src/content/1.7/modules/creation/adding-configuration-page.md:
````diff
@@ -320,3 +320,100 @@ in the back office:
+],
+```
+
+### select example
````
```suggestion
### Select example
```

Progi1984 reviewed (commented):
Some feedbacks

jf-viguier committed:
typo

Co-authored-by: Progi1984 <progi1984@gmail.com>

jf-viguier committed:
typo

Co-authored-by: Progi1984 <progi1984@gmail.com>

jf-viguier committed:
Typo

Co-authored-by: Progi1984 <progi1984@gmail.com>

jf-viguier committed:
more clear

jf-viguier committed:
Add validation

jf-viguier committed:
short array notation

Progi1984 commented:
@jf-viguier Could you check your modification in the PR #1074 (from @eternoendless) ?

jf-viguier commented:
@Progi1984 my version of HelperForm is better : more complete, array notation, exemples.
If you merge  #1074 I can move this par here.

Sinepel reviewed (approved):
Thanks @jf-viguier !

jf-viguier committed:
add legacy part

eternoendless commented:
Now that #1074 has been merged, would you kindly update your PR so that the content is on the right page? Thanks!

matks commented:
Hello,

I have modified your PR to target the `1.7.x` branch.

We have modified the devdocs inner working in order to welcome multiple documentation versions. This is because soon we will have a 1.7 documentation and a 8.0 documentation, matching PrestaShop 1.7 or PrestaShop 8.0 softwares.

Read more on https://devdocs.prestashop.com/1.7/contribute/documentation/how/documentation-versions/

Please also note that now that we have multiple versions inside one documentation, we had to build a workflow that can handle this setup. This repository https://github.com/PrestaShop/docs now only contains the content of the documentation, and https://github.com/PrestaShop/devdocs-site contains the sources and the deployment workflow.

Because of this, some files have moved and you might have to rebase your Pull Request to match the new structure of the repository.

matks committed:
Explain Form Theme and project progress

Commit:
Feedbacks

Commit:
Add compatibility table

Commit:
Modifiy compatiblity table

Commit:
Correct typo

Commit:
Change table ordfer

matks committed:
Apply suggestions from code review

Co-authored-by: Pablo Borowicz <pablo.borowicz@prestashop.com>

matks committed:
Update _index.md

matks committed:
Update _index.md

Commit:
Remove weight from component pages so that they are sorted alphabetically

matks committed:
Merge pull request #1121 from eternoendless/remove-weight-cmp

Remove weight from component pages so that they are sorted alphabetically

maxime-aknin committed:
fix: broken image link

matthieu-rolland committed:
Merge pull request #1123 from maxime-aknin/patch-1

fix: broken image link

matks committed:
Update overriding-modules.md

Thymotep committed:
Update good-practices.md

Added the prohibition to use global variables in modules, in accordance with the validator rules.

matks committed:
Apply suggestions from code review

matks committed:
Merge pull request #1127 from Thymotep/patch-1

Update good module practices

kpodemski committed:
Merge pull request #1126 from matks/matks-patch-1

Improve a little overriding module page

WebHelpersPau committed:
Update tools-for-theme-designers.md

PierreRambaud committed:
Merge pull request #1128 from WebHelpersPau/patch-4

Update tools-for-theme-designers.md

mreduar committed:
Updated .php-cs-fixer and pre-commit links

PierreRambaud committed:
Merge pull request #1132 from mreduar/patch-1

Updated .php-cs-fixer and pre-commit links

matks committed:
Apply suggestions from code review

Co-authored-by: Pablo Borowicz <pablo.borowicz@prestashop.com>

Commit:
Refacto struggling sentence

Co-authored-by: Pablo Borowicz <pablo.borowicz@prestashop.com>

Commit:
Refacto node-gyp misspelling

Co-authored-by: Pablo Borowicz <pablo.borowicz@prestashop.com>

Commit:
Review feedback of table versions

Co-authored-by: Pablo Borowicz <pablo.borowicz@prestashop.com>

Commit:
Update development/compile-assets.md

Co-authored-by: Pablo Borowicz <pablo.borowicz@prestashop.com>

Commit:
Refacto table list title

kpodemski committed:
additional info in the webservice example

Progi1984 committed:
Merge pull request #1134 from kpodemski/webservice-sort

Additional info in the webservice example

matks committed:
Use ffittschen/pr-branch-labeler

PierreRambaud committed:
Missing end of line at the end of pr-branch-labeler.yml file

PierreRambaud committed:
Add missing end of line in pr-branch-labeler.yml file

PierreRambaud committed:
Merge pull request #1144 from matks/label-prs-17x

Use ffittschen/pr-branch-labeler (1.7.x)

SebSept committed:
wrong autoload file path

case matters on GNU linux (at least).

matks committed:
Merge pull request #1149 from SebSept/patch-1

Webservice lib : fix wrong autoload file path

matks committed:
Fix typo in software licenses page

TheDreamCatcher committed:
Fix Position location

zou-simon committed:
Update classic-system.md

Line 361, change "without the extension removed" to "with the extension removed"

matks committed:
Merge pull request #1154 from TheDreamCatcher/patch-1

Fix Position location

matks committed:
Update pr-branch-labeler.yml

PierreRambaud committed:
Merge pull request #1157 from PrestaShop/matks-patch-2

Update pr-branch-labeler to use pull_request_target on 1.7.x branch

Commit:
Update npm versions needed

StephaneBullier committed:
Typo correction

PierreRambaud committed:
Merge pull request #1055 from NeOMakinG/update-node

Precise nodejs version, how to switch and node-gyp dependency on Windows

PierreRambaud committed:
Merge pull request #1160 from StephaneBullier/patch-1

Typo correction

SebSept committed:
fix 2 tiny typo

legacy + correct form for a.k.a. (final dot)

kpodemski committed:
Merge pull request #1161 from SebSept/patch-3

fix 2 tiny typo

juangiordana committed:
Correct spelling errors.

Correct spelling errors.

juangiordana committed:
Shorten `location` regex for browser caching.

Shorten and sort values alphabetically in `location` regex for browser caching.

juangiordana committed:
Improve documentation to enable compression in Nginx.

Improve recommendations to enable compression in Nginx.

http://nginx.org/en/docs/http/ngx_http_gzip_module.html

* Update `gzip_min_length` value to 1000. This is is the suggested value used in the [ngx_http_gzip_module](http://nginx.org/en/docs/http/ngx_http_gzip_module.html) page.

* Updated `gzip_types` to the most commonly used values, sorted alphabetically.

* Removed `gzip_http_version 1.1` since it's the default.

* Removed `gzip_buffers` since sane (and same) values are provided by default.

* Removed `gzip_static` directive since it's not commonly used and pollutes the provided example.

juangiordana committed:
Remove gzip settings

Remove gzip settings from the main Nginx configuration example as its
usage may vary on different HTTP/HTTPS scenarios and provide a link to
the internal Nginx optimization article that contains improved gzip
settings.

juangiordana committed:
Enable HTTPS by default.

Change default `listen` directives so HTTPS is enabled by default as
it's the most common configuration nowadays, specially for e-commerce
websites.

juangiordana committed:
Remove additional SSL configuration block.

Remove additional SSL configurations that pollutes the example and may
vary across different server configurations.

juangiordana committed:
Remove duplicate `server_name` entry.

Remove duplicate `server_name` entry.

juangiordana committed:
Add `access_log` and `error_log` directives.

Add `access_log` and `error_log` directives.

juangiordana committed:
Add `nginx` syntax highlighting to code blocks.

Add markdown `nginx` syntax highlighting to Nginx code blocks.

juangiordana committed:
Update list of denied directories.

Update list of denied directories.

- Add `var`.
- Remove `tests-legacy`.

matks committed:
Merge pull request #1155 from zou-simon/patch-1

Fix Classic translation system documentation

juangiordana committed:
Improve comments in `nginx.conf`.

Improve comments in the `nginx.conf` example.

juangiordana committed:
Sort regex alphabetically.

Sort regex parameters alphabetically.

juangiordana committed:
Add default `location / {...}` block.

Add default `location / {...}` block and move `try_files` directive
from a `server` context to this default `location` context.

> The try_files directive is mostly like every other action directive
> mentioned above, the difference is that if placed in server context
> nginx actually creates a pseudo-location that is the least specific
> location possible. That means if a request matches a defined location
> the try_files directive will not be executed. This means that if you
> have location / defined then you have a location that matches every
> possible request and as such try_files will never actually execution.
> Therefore always place try_files in location context instead of
> server context if at all possible.

[Understanding the Nginx Configuration Inheritance Model](https://blog.martinfjordvald.com/understanding-the-nginx-configuration-inheritance-model/).

Use `$is_args$args` in the `try_files` directive as suggested in the
[Symfony documentation](https://symfony.com/doc/current/setup/web_server_configuration.html#web-server-nginx).

juangiordana committed:
Sort AlphaImageLoader regex parameters alphabetically.

Sort AlphaImageLoader regex parameters alphabetically.

juangiordana committed:
Update list of denied directories.

Update list of denied directories.

- Remove `tools`.
- Remove `travis-scripts`.

Co-authored-by: GoT <PierreRambaud@users.noreply.github.com>

juangiordana committed:
Simplify `location` regex.

Simplify "Cloudflare / Max CDN fix." `location` regex.

Co-authored-by: GoT <PierreRambaud@users.noreply.github.com>

juangiordana committed:
Improve `location` regex for browser caching.

Improve `location` regex for browser caching.

Improve `location` regex for browser caching by adding the following file extensions: `eot`, `otf`, `ttf`, `woff`, `woff2`.

Co-authored-by: GoT <PierreRambaud@users.noreply.github.com>

PierreRambaud commented:
ping @jf-viguier  friendly reminder :heart:

juangiordana committed:
Add "Access logging" section.

Add "Access logging" section to the "Optimize your Nginx configuration"
article.

Remove these directives from the `nginx.conf` configuration example as
this may vary on different PrestaShop scenarios and pollute the
simplicity of the provided example.

juangiordana committed:
Remove "Cloudflare / Max CDN fix" from `nginx.conf`

Remove "Cloudflare / Max CDN fix" from `nginx.conf` as its use is not
required in a basic PrestaShop installation.

juangiordana committed:
Remove "Force download of ..." sections from `nginx.conf`

Remove "Force download of ..." sections from the main `nginx.conf`
example as their usage may vary across different PrestaShop
installations and hinder the simplicity of the provided example, i.e.,
the file already provides a `location /upload` section and these
directives could be specified in that section.

PierreRambaud committed:
Deny tools directory

matthieu-rolland committed:
Merge pull request #1166 from juangiordana/1.7.x-nginx

1.7.x nginx

matks committed:
Add all SF forms to form theme page

matks committed:
Move form theme img to their directory

Commit:
Merge pull request #1118 from matks/document-form-simplification-2

PierreRambaud committed:
Merge pull request #1174 from matks/backport-1153

Backport PR 1153 to 1.7.x

Commit:
Fix slack link

jf-viguier committed:
Move helperOptions stuff to helperoptions.md

jf-viguier committed:
Merge branch 'patch-6' of github.com:jf-viguier/docs into patch-6

# Conflicts:
#	src/content/1.7/modules/creation/adding-configuration-page.md

jf-viguier commented:
@PierreRambaud @matks Rebase stuff is broken, I've created a new pr here https://github.com/PrestaShop/docs/pull/1185

Progi1984 commented:
@jf-viguier I close this one.
